wradlib/io/dem.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# Copyright (c) 2011-2019, wradlib developers.
+# Distributed under the MIT License. See LICENSE.txt for more info.
+
+
+"""
+Digital elevation model I/O
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Provide surface/terrain elevation information
+
+.. autosummary::
+   :nosignatures:
+   :toctree: generated/
+
+download_srtm
+get_srtm
+
+"""
+
+import numpy as np
+import os
+
+from osgeo import gdal
+import requests
+
+import wradlib.util
+
+
+def download_srtm(filename, destination=None, version=2, resolution=3,
+                  region="Eurasia", token=None):
+    """
+    Download NASA SRTM elevation data
+    Version 3 is only available with a login and a token
+
+    Parameters
+    ----------
+    filename : str
+        srtm file to download
+    destination : str
+        output filename
+    version : int
+        srtm version (2 or 3)
+    resolution : int
+        resolution of SRTM data (1, 3 or 30)
+    region : str
+        name of the region for SRTM version 2 only:
+        Africa, Australia, Eurasia, Islands, North America, South America
+    token : str
+        filename with authorization token (required for version 3)
+
+    """
+    if version == 2:
+        website = "https://dds.cr.usgs.gov/srtm/version2_1"
+        source = "%s/SRTM%s/%s" % (website, resolution, region)
+    if version == 3:
+        website = "http://e4ftl01.cr.usgs.gov/MEASURES"
+        source = "%s/SRTMGL%d.003/2000.02.11" % (website, resolution)
+    url = "%s/%s" % (source, filename)
+
+    headers = None
+    if token is not None:
+        headers = {'Authorization': 'Bearer %s' % (token)}
+    try:
+        r = requests.get(url, headers=headers, stream=True)
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as err:
+        print(err)
+        return
```

Comment:
We need to think about a better solution here. Just swallowing the exception here doesn't seem the right way IMHO.

Simplest thing would be to add a specifying wradlib message to the error message and re-raise.

Reply:
Actually it has to pass if the HTTP error is 404 because it is expected that "sea" files are missing. Other HTTP errors should reraise.

Reply:
> What about the coverage of the download function which seems difficult to test?

We could try to test, but decorate it with "knownfail"?